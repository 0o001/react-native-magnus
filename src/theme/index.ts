export { ThemeContext } from './themeContext';
export { withTheme } from './withTheme';
export { ThemeProvider, ThemeProviderProps } from './themeProvider.component';

export {
  ThemeType,
  BorderPropsType,
  SpacingPropsType,
  RoundedPropsType,
} from './type';
