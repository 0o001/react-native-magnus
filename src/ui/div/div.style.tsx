import { StyleSheet } from "react-native";

import {
  getThemeProperty,
  createFlexStyles,
  createPositionStyle,
  createSpacingStyles,
  createBorderWidthStyles,
  createBorderColorStyles,
  createBorderRadiusStyles
} from "../../theme/theme.service";

/**
 * computed style
 *
 * @param theme
 * @param props
 */
export const getStyle = (theme: any, props: any) => {
  const computedStyle: any = {};

  computedStyle.div = {
    flexDirection: props.flexDir,
    flexWrap: props.flexWrap,
    alignItems: props.alignItems,
    justifyContent: props.justifyContent,
    alignSelf: props.alignSelf,
    backgroundColor: getThemeProperty(theme.colors, props.bg, "transparent"),
    ...createFlexStyles(props),
    ...createPositionStyle(props),
    ...createBorderWidthStyles(props),
    ...createSpacingStyles(props, theme.spacing),
    ...createBorderColorStyles(props, theme.colors),
    ...createBorderRadiusStyles(props, theme.borderRadius)
  };

  computedStyle.image = {
    ...createBorderRadiusStyles(props, theme.borderRadius)
  };

  if (props.shadow) {
    computedStyle.div = {
      ...computedStyle.div,
      ...theme.shadow[props.shadow],
      shadowColor: getThemeProperty(theme.colors, props.shadowColor, "white")
    };
  }

  if (props.h) {
    computedStyle.div = {
      ...computedStyle.div,
      height: props.h
    };
  }

  if (props.w) {
    computedStyle.div = {
      ...computedStyle.div,
      width: props.w
    };
  }

  if (props.minH) {
    computedStyle.div = {
      ...computedStyle.div,
      minHeight: props.minH
    };
  }

  if (props.minW) {
    computedStyle.div = {
      ...computedStyle.div,
      minWidth: props.minW
    };
  }

  if (props.opacity) {
    computedStyle.div = {
      ...computedStyle.div,
      opacity: props.opacity
    };
  }

  // merging style props to computed style
  if (props.style) {
    computedStyle.div = {
      ...computedStyle.div,
      ...props.style
    };
  }

  return StyleSheet.create(computedStyle);
};
