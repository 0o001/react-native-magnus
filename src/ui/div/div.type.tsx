import {
  ViewProps as RNViewProps,
  ImageSourcePropType as RNImageSourcePropType,
} from 'react-native';

import {
  BorderPropsType,
  SpacingPropsType,
  RoundedPropsType,
  ShadowPropsType,
  ResponsiveType,
  BaseBgModeType,
  BaseJustifyType,
  BaseAlignSelfType,
  BaseAlignItemsType,
  BaseFlexWrapType,
  BaseFlexDirectionType,
  BasePositionType,
  BaseOverflowType,
  BaseBorderStyleType,
} from '../../theme';

export interface DivProps
  extends RNViewProps,
    BorderPropsType,
    SpacingPropsType,
    RoundedPropsType,
    ShadowPropsType {
  h?: ResponsiveType<string | number>;
  w?: ResponsiveType<string | number>;
  bg?: ResponsiveType<string>;
  minH?: ResponsiveType<string | number>;
  minW?: ResponsiveType<string | number>;
  maxH?: ResponsiveType<string | number>;
  maxW?: ResponsiveType<string | number>;
  bgImg?: RNImageSourcePropType;
  bgMode?: ResponsiveType<BaseBgModeType>;
  flex?: ResponsiveType<number>;
  justifyContent?: ResponsiveType<BaseJustifyType>;
  alignSelf?: ResponsiveType<BaseAlignSelfType>;
  alignItems?: ResponsiveType<BaseAlignItemsType>;
  flexDir?: ResponsiveType<BaseFlexDirectionType>;
  flexWrap?: ResponsiveType<BaseFlexWrapType>;
  position?: ResponsiveType<BasePositionType>;
  overflow?: ResponsiveType<BaseOverflowType>;
  borderStyle?: ResponsiveType<BaseBorderStyleType>;
  top?: ResponsiveType<number>;
  right?: ResponsiveType<number>;
  bottom?: ResponsiveType<number>;
  left?: ResponsiveType<number>;
  opacity?: ResponsiveType<number>;
  children?: React.ReactNode[] | React.ReactNode;
  row?: ResponsiveType<boolean>;
  zIndex?: ResponsiveType<number>;
}
