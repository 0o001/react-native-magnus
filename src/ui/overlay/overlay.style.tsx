import { StyleSheet } from "react-native";

import {
  getThemeProperty,
  createFlexStyles,
  createSpacingStyles,
  createBorderRadiusStyles
} from "../../theme/theme.service";

/**
 * computed style
 *
 * @param theme
 * @param props
 */
export const getStyle = (theme: any, props: any) => {
  const computedStyle: any = {};

  computedStyle.modal = {
    backgroundColor: "rgba(0, 0, 0, 0.6)",
    flex: 1,
    alignItems: "center",
    justifyContent: "center"
  };

  computedStyle.container = {
    alignSelf: "center",
    backgroundColor: getThemeProperty(theme.colors, props.bg, "transparent"),
    ...createFlexStyles(props),
    ...createSpacingStyles(props, theme.spacing),
    ...createBorderRadiusStyles(props, theme.borderRadius),
    width: props.w,
    height: props.h,
    alignItems: props.alignItems,
    justifyContent: props.justifyContent
  };

  // merging style props to computed style
  if (props.style) {
    computedStyle.container = {
      ...computedStyle.container,
      ...props.style
    };
  }

  return StyleSheet.create(computedStyle);
};
