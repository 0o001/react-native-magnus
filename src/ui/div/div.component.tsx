import * as React from "react";
import { useContext } from "react";
import {
  View as RNView,
  ImageBackground as RNImageBackground
} from "react-native";

import { DivProps } from "./div.type";
import { getStyle } from "./div.style";
import { ThemeContext } from "../../theme";

const Div: React.FunctionComponent<DivProps> = (props: DivProps) => {
  const {
    bg,
    h,
    w,
    m,
    mt,
    mr,
    mb,
    ml,
    p,
    pr,
    pt,
    pb,
    pl,
    minH,
    minW,
    maxW,
    maxH,
    position,
    style,
    flexDir,
    row,
    rounded,
    roundedTop,
    roundedRight,
    roundedBottom,
    roundedLeft,
    children,
    bgImg,
    bgMode,
    alignItems,
    justifyContent,
    borderColor,
    borderBottomColor,
    borderLeftColor,
    borderTopColor,
    borderRightColor,
    borderWidth,
    borderLeftWidth,
    borderRightWidth,
    borderBottomWidth,
    borderTopWidth,
    flexWrap,
    shadow,
    shadowColor,
    opacity,
    overflow,
    top,
    left,
    right,
    bottom,
    zIndex,
    ...rest
  } = props;
  const theme = useContext(ThemeContext);
  const computedStyle = getStyle(theme, props);

  if (bgImg) {
    return (
      <RNImageBackground
        source={bgImg}
        style={computedStyle.div}
        resizeMode={props.bgMode}
        imageStyle={computedStyle.image}
        {...rest}
      >
        {children}
      </RNImageBackground>
    );
  }

  return (
    <RNView style={computedStyle.div} {...rest}>
      {children}
    </RNView>
  );
};

Div.defaultProps = {
  bg: "transparent",
  flexDir: "column",
  flexWrap: "nowrap",
  rounded: "none",
  shadow: 0,
  shadowColor: "gray900",
  position: "relative",
  bgMode: "cover",
  pointerEvents: "auto",
  row: false,
  borderStyle: "solid"
};

export { Div };
