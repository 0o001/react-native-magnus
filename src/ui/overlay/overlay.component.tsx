import * as React from "react";
import { useContext } from "react";
import { View as RNView, Modal as RNModal } from "react-native";

import { getStyle } from "./overlay.style";
import { ThemeContext } from "../../theme";
import { OverlayProps } from "./overlay.type";

const Overlay: React.FunctionComponent<OverlayProps> = (
  props: OverlayProps
) => {
  const {
    m,
    mt,
    mr,
    mb,
    ml,
    p,
    pr,
    pt,
    pb,
    pl,
    bg,
    w,
    alignItems,
    justifyContent,
    rounded,
    roundedTop,
    roundedRight,
    roundedBottom,
    roundedLeft,
    visible,
    children,
    overlayColor,
    overlayOpacity,

    ...rest
  } = props;
  const theme = useContext(ThemeContext);
  const computedStyle = getStyle(theme, props);

  return (
    <RNModal
      animationType="fade"
      transparent
      visible={visible}
      onRequestClose={() => {}}
      {...rest}
    >
      <RNView style={computedStyle.modal}>
        <RNView style={computedStyle.container}>{children}</RNView>
      </RNView>
    </RNModal>
  );
};

Overlay.defaultProps = {
  bg: "white",
  w: "80%",
  p: "lg",
  rounded: "md",
  visible: false,
  overlayOpacity: 0.6,
  overlayColor: "gray900"
};

export { Overlay };
