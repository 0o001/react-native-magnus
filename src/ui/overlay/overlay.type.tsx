import { ModalProps as RNModalProps } from 'react-native';

import { SpacingPropsType, RoundedPropsType } from '../../theme';

export interface OverlayRef {
  close: any;
  open: any;
}

export interface OverlayProps
  extends RNModalProps,
    SpacingPropsType,
    RoundedPropsType {
  bg?: string;
  w?: string | number;
  h?: number | string;
  flex?: number;
  minH?: number | string;
  minW?: number | string;
  justifyContent?:
    | 'flex-start'
    | 'flex-end'
    | 'center'
    | 'space-between'
    | 'space-around'
    | 'space-evenly';
  alignItems?: 'flex-start' | 'flex-end' | 'center' | 'stretch' | 'baseline';
  flexDir?: 'row' | 'column' | 'row-reverse' | 'column-reverse';
  flexWrap?: 'wrap' | 'nowrap' | 'wrap-reverse';
  children: React.ReactElement[] | React.ReactElement;
  isVisible?: boolean;
  overlayColor?: string;
  overlayOpacity?: number;
}
