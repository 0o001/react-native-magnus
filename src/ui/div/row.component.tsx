import * as React from "react";
import { useContext } from "react";
import {
  View as RNView,
  ImageBackground as RNImageBackground
} from "react-native";

import { DivProps } from "./div.type";
import { getStyle } from "./div.style";
import { ThemeContext } from "../../theme";

const Row: React.FunctionComponent<DivProps> = (props: DivProps) => {
  const {
    h,
    w,
    m,
    mt,
    mr,
    mb,
    ml,
    p,
    pr,
    pt,
    pb,
    pl,
    minH,
    minW,
    position,
    style,
    flexDir,
    rounded,
    roundedTop,
    roundedRight,
    roundedBottom,
    roundedLeft,
    children,
    bgImg,
    bgMode,
    alignItems,
    justifyContent,
    borderColor,
    borderBottomColor,
    borderLeftColor,
    borderTopColor,
    borderRightColor,
    borderWidth,
    borderLeftWidth,
    borderRightWidth,
    borderBottomWidth,
    borderTopWidth,
    flexWrap,
    shadow,
    shadowColor,
    ...rest
  } = props;
  const theme = useContext(ThemeContext);
  const computedStyle = getStyle(theme, props);

  if (bgImg) {
    return (
      <RNImageBackground
        source={bgImg}
        style={computedStyle.div}
        resizeMode={props.bgMode}
        imageStyle={computedStyle.image}
        {...rest}
      >
        {children}
      </RNImageBackground>
    );
  }

  return (
    <RNView style={computedStyle.div} {...rest}>
      {children}
    </RNView>
  );
};

Row.defaultProps = {
  bg: "white",
  flexDir: "row",
  flexWrap: "nowrap",
  rounded: "none",
  alignItems: "center",
  shadow: 0,
  shadowColor: "gray900",
  position: "relative",
  bgMode: "cover"
};

export { Row };
