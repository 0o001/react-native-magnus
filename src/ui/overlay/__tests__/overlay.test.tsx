import React from "react";
import renderer from "react-test-renderer";

import { Overlay } from "../overlay.component";
import { Div } from "../../div/div.component";
import { ThemeProvider } from "../../../theme";

describe("Overlay Component", () => {
  it("should render without issues", () => {
    const instance = renderer.create(
      <ThemeProvider>
        <Overlay visible>
          <Div />
        </Overlay>
      </ThemeProvider>
    );

    expect(instance.toJSON()).toMatchSnapshot();
  });
});
