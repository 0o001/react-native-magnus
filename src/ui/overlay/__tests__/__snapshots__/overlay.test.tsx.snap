// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Overlay Component should render without issues 1`] = `
<Modal
  animationType="fade"
  hardwareAccelerated={false}
  onRequestClose={[Function]}
  transparent={true}
  visible={true}
>
  <View
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "rgba(0, 0, 0, 0.6)",
        "flex": 1,
        "justifyContent": "center",
      }
    }
  >
    <View
      style={
        Object {
          "alignItems": "center",
          "alignSelf": "center",
          "backgroundColor": "#FFFFFF",
          "borderRadius": 6,
          "height": undefined,
          "justifyContent": "center",
          "padding": 12,
          "width": "80%",
        }
      }
    >
      <View
        style={
          Object {
            "alignItems": undefined,
            "backgroundColor": "#FFFFFF",
            "borderRadius": 0,
            "flexDirection": "column",
            "flexWrap": "nowrap",
            "justifyContent": undefined,
            "position": "relative",
          }
        }
      />
    </View>
  </View>
</Modal>
`;
